File: RepoTransBench/target_projects/Python/C++/ecmwf-lab_ai-models/public_tests/test_public_main_basic_test.cpp
#include <gtest/gtest.h>
#include <stdexcept>
#include <string>
#include <vector>
#include "src/ai_models/main.h"

Args dummy_parse_args(const std::vector<std::string>& argv) {
    if (std::find(argv.begin(), argv.end(), "nonsense_command") != argv.end()) {
        throw SystemExitException(2);
    }
    Args args;
    args.command = argv.at(0);
    args.dummy = argv.size() > 2 ? argv.at(2) : "";
    return args;
}

class DummyParseArgsContext {
public:
    DummyParseArgsContext() {
        set_parse_args(dummy_parse_args);
    }
    ~DummyParseArgsContext() {
        reset_parse_args();
    }
};

TEST(PublicMain, ParseArgs) {
    DummyParseArgsContext ctx;
    std::vector<std::string> argv = {"run", "--dummy", "xy"};
    auto args = parse_args(argv);
    EXPECT_EQ(args.command, "run");
    EXPECT_TRUE(args.dummy == "xy" || args.dummy == "");
}

TEST(PublicMain, InvalidArgs) {
    DummyParseArgsContext ctx;
    std::vector<std::string> argv = {"nonsense_command"};
    EXPECT_THROW(parse_args(argv), SystemExitException);
}

File: RepoTransBench/target_projects/Python/Go/ecmwf-lab_ai-models/public_tests/test_public_main_basic_test.go
package public_tests

import (
	"errors"
	"strings"
	"testing"
)

type DummyArgs struct {
	Command string
	Dummy   string
}

func dummyParseArgs(argv []string) (*DummyArgs, error) {
	var dummy string
	if len(argv) > 2 {
		dummy = argv[2]
	}
	if len(argv) > 0 && argv[0] == "nonsense_command" {
		return nil, errors.New("SystemExit")
	}
	return &DummyArgs{
		Command: argv[0],
		Dummy:   dummy,
	}, nil
}

func TestPublicMainParseArgs(t *testing.T) {
	argv := []string{"run", "--dummy", "xy"}
	args, err := dummyParseArgs(argv)
	if err != nil {
		t.Fatalf("parse_args failed: %v", err)
	}
	if args.Command != "run" {
		t.Errorf("args.Command = %v, want run", args.Command)
	}
	if args.Dummy != "xy" && args.Dummy != "" {
		t.Errorf("args.Dummy = %v, want xy or empty", args.Dummy)
	}
}

func TestPublicMainInvalidArgs(t *testing.T) {
	argv := []string{"nonsense_command"}
	_, err := dummyParseArgs(argv)
	if err == nil {
		t.Errorf("expected SystemExit (error), got nil")
	}
}